Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp

```diff
@@ -113,36 +119,118 @@ namespace AzToolsFramework
         return returnedNode;
     }
 
+    DPEModelNode* DPEModelNode::GetChildFromDomIndex(size_t domIndex)
+    {
+        // search children for node that has the given dom index
+        // These are almost always tiny lists, so using a Map would be overkill and almost certainly slower in practice
+        DPEModelNode* returnedNode = nullptr;
+
+        // search row and column children for the node with the given DOM index
+        for (auto rowIterator = m_rowChildren.begin(), endIterator = m_rowChildren.end(); rowIterator != endIterator; ++rowIterator)
+        {
+            if ((*rowIterator)->GetDomValueIndex() == domIndex)
+            {
+                returnedNode = *rowIterator;
+                break;
+            }
+        }
+        for (auto columnIterator = m_columnChildren.begin(), endIterator = m_columnChildren.end(); columnIterator != endIterator;
+             ++columnIterator)
+        {
+            if ((*columnIterator)->GetDomValueIndex() == domIndex)
+            {
+                returnedNode = *columnIterator;
+                break;
+            }
+        }
+
+        return returnedNode;
+    }
+
     int DPEModelNode::RowOfChild(DPEModelNode* const childNode) const
     {
-        return m_children.indexOf(childNode);
+        return m_rowChildren.indexOf(childNode);
     }
 
-    void DPEModelNode::Populate(const AZ::Dom::Value& domVal)
+    int DPEModelNode::ColumnOfChild(DPEModelNode* const childNode) const
     {
-        auto& jsonBackend = GetModel()->GetBackend();
-        for (size_t arrayIndex = 0, numIndices = domVal.ArraySize(); arrayIndex < numIndices; ++arrayIndex)
+        return m_columnChildren.indexOf(childNode);
+    }
+
+    void DPEModelNode::SetValue(const AZ::Dom::Value& domVal, bool notifyView)
+    {
+        AZStd::string domName = domVal.GetNodeName().GetCStr();
```

Comment:
nit: can store this as `AZStd::string_view` (and use `GetStringVIew`)

Reply:
yep! Done :)